python_modules/dagster/dagster/core/storage/fs_asset_io_manager.py

```diff
@@ -81,4 +81,8 @@ def asset2(asset1):
 
 class AssetPickledObjectFilesystemIOManager(PickledObjectFilesystemIOManager):
     def _get_path(self, context):
-        return os.path.join(self.base_dir, *context.get_asset_output_identifier())
+        if context.has_asset_key:
+            path = context.get_asset_output_identifier()
+        else:
+            path = [context.run_id, context.step_key]
```

Comment:
nit: can this be context.get_run_scoped_asset_identifier?

Reply:
Thanks! didn't realize that function existed